test/logic_value_test.dart

```diff
@@ -947,4 +949,44 @@ void main() {
           equals(LogicValue.filled(0, LogicValue.zero).hashCode));
     });
   });
+
+  group('random value generation ', () {
+    test(
+        'should return random integer logic value if min and max value '
+        'is in integer', () {
+      final lvInt = Random(10).nextLogicValueInt(min: 20, max: 100);
+
+      expect(lvInt.toInt(), isA<int>());
+      expect(lvInt, isA<LogicValue>());
+    });
+
+    test('should return random BigInt value', () {
+      final bigInt = Random(10).nextBigInt(numBits: 100);
+
+      expect(bigInt, isA<BigInt>());
+      expect(bigInt.bitLength, 100);
+    });
+
+    test('should return random BigInt logic value', () {
+      final lvBigInt = Random(10).nextLogicValueBigInt(numBits: 100);
+
+      expect(lvBigInt, isA<LogicValue>());
+      expect(lvBigInt.toBigInt(), isA<BigInt>());
+      expect(lvBigInt.width, 100);
+    });
+    test(
+        'should return random logic value with invalid bits '
+        'if hasInvalidBits is true', () {
+      final lvRand = Random().nextLogicValue(width: 10, hasInvalidBits: true);
+
+      print(lvRand);
+    });
+
+    test(
+        'should return random logic value without invalid bits '
+        'if hasInvalidBits is false', () {
+      final lvRand = Random().nextLogicValue(width: 10, max: 20);
+      print(lvRand);
+    });
+  });
```

Comment:
add tests that exercise different number of bits (0, 1-31, 32, 31-63, 64, >64) to hit edge cases.